### PR TITLE
Allow usage of rpmsg static freertos related api

### PR DESCRIPTION
**Pull request by Hadatko**
This change depends on https://github.com/NXPmicro/rpmsg-lite/pull/14

### CODE REVIEW

File: erpc_c/transports/erpc_rpmsg_lite_base_transport.h [hunk 1]
```diff
@@ -29,6 +30,9 @@
 // Classes
 ////////////////////////////////////////////////////////////////////////////////
 
+/* TODO: replace bellow size representing sizeof(rpmsg_queue_rx_cb_data_t)*/
+#define SIZE_OF_RPMSG_QUEUE 12
```
Comment (Hadatko) [inline erpc_c/transports/erpc_rpmsg_lite_base_transport.h:34]: @MichalPrincNXP we need get rid of this macro. Any idea?

**Comment by MichalPrincNXP** [comment]
Hi Dusan, let's target these changes for the next release (after v1.8.1 release), ok? Thanks.

**Comment by Hadatko** [comment]
Hi @MichalPrincNXP , yes sure ;)

File: erpc_c/transports/erpc_rpmsg_lite_base_transport.h (hunk 1 above)
Comment (MichalPrincNXP) [reply erpc_c/transports/erpc_rpmsg_lite_base_transport.h:42]: There could be 
#define RL_ENV_QUEUE_STATIC_STORAGE_SIZE (RL_BUFFER_COUNT * sizeof(rpmsg_queue_rx_cb_data_t))
placed in rpmsg_env_specific.h 

and then we do not need the SIZE_OF_RPMSG_QUEUE macro and instead we could use 
uint8_t m_queue_stack[RL_ENV_QUEUE_STATIC_STORAGE_SIZE];
in RPMsgBaseTransport class

I am just testing the static allocation update in rpmsg_lite and once it is stable we can focus on erpc adaptation.

File: erpc_c/transports/erpc_rpmsg_lite_rtos_transport.cpp [hunk 2]
```diff
@@ -137,7 +149,11 @@ erpc_status_t RPMsgRTOSTransport::init(uint32_t src_addr, uint32_t dst_addr, voi
 
     if (0U == s_initialized)
     {
+#if RL_USE_STATIC_API s_rpmsg =
```
Comment (MichalPrincNXP) [inline erpc_c/transports/erpc_rpmsg_lite_rtos_transport.cpp:152]: typo (s_rpmsg =)

**Comment by MichalPrincNXP** [comment]
#if ERPC_ALLOCATION_POLICY == ERPC_STATIC_POLICY
macro names need to be aligned with #169

**Comment by Hadatko** [comment]
Sure, will do

File: erpc_c/transports/erpc_rpmsg_lite_rtos_transport.cpp (hunk 2 above)
Comment (Hadatko) [reply erpc_c/transports/erpc_rpmsg_lite_rtos_transport.cpp:152]: hmm interesting. Will fix

**Comment by Hadatko** [comment]
Fixes pushed. Is that ok now?

File: erpc_c/port/erpc_setup_extensions_freertos.cpp [hunk 3]
```diff
@@ -16,6 +16,9 @@ using namespace erpc;
 
 static Semaphore *s_erpc_call_in_progress = NULL;
 static TimerHandle_t s_erpc_call_timer_cb = NULL;
+#if configSUPPORT_STATIC_ALLOCATION
```
Comment (MichalPrincNXP) [inline erpc_c/port/erpc_setup_extensions_freertos.cpp:19]: Condition in contradiction with #169 , should be changed to the way used in #169, i.e.
#if ERPC_ALLOCATION_POLICY == ERPC_ALLOCATION_POLICY_STATIC

applicable on more places

File: erpc_c/transports/erpc_rpmsg_lite_base_transport.h [hunk 4]
```diff
@@ -54,6 +66,12 @@ class RPMsgBaseTransport : public Transport
 protected:
     static struct rpmsg_lite_instance *s_rpmsg; /*!< Pointer to instance of RPMSG lite. */
     static uint8_t s_initialized;               /*!< Represent information if the rpmsg-lite was initialized. */
+#if RL_USE_STATIC_API
+    struct rpmsg_lite_instance m_static_context;
+    struct rpmsg_lite_ept_static_context m_ept_context;
+    rpmsg_static_queue m_queue_context;
```
Comment (MichalPrincNXP) [inline erpc_c/transports/erpc_rpmsg_lite_base_transport.h:72]: rpmsg_static_queue -> rpmsg_static_queue_ctxt as per latest changes in the rpmsg_lite repo

File: erpc_c/transports/erpc_rpmsg_lite_base_transport.h (hunk 1 above)
Comment (MichalPrincNXP) [reply erpc_c/transports/erpc_rpmsg_lite_base_transport.h:42]: SIZE_OF_RPMSG_QUEUE can be removed

File: erpc_c/transports/erpc_rpmsg_lite_base_transport.h [hunk 5]
```diff
@@ -54,6 +66,12 @@ class RPMsgBaseTransport : public Transport
...
+    uint8_t m_queue_stack[SIZE_OF_RPMSG_QUEUE * RL_BUFFER_COUNT];
```
Comment (MichalPrincNXP) [inline erpc_c/transports/erpc_rpmsg_lite_base_transport.h:73]: [SIZE_OF_RPMSG_QUEUE * RL_BUFFER_COUNT]
->
[RL_ENV_QUEUE_STATIC_STORAGE_SIZE]

as per latest changes in the rpmsg_lite repo

**Comment by Hadatko** [comment]
Update pushed.

**Review by MichalPrincNXP — APPROVED** [review]
This PR is linked with #169 and will be integrated together. Nevertheless, this does not allow static allocation of all erpc objects and the work on the static allocations needs to continue (information for others looking at this PR). "Known issues and limitations" section should be added into the wiki with following statement:
Static allocations controlled by the ERPC_ALLOCATION_POLICY config macro are not fully supported yet, i.e. not all erpc objects can be allocated statically now. It deals with the ongoing process and the full static allocations support will be added in the future.

**Comment by MichalPrincNXP** [comment]
Thank you for this effort!

**Pull request merged**